core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -608,9 +613,9 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
    * Request that the cluster manager kill all executors on a given host.
    * @return whether the kill request is acknowledged
    */
-  final override def killExecutorsOnHost(host: String): Seq[String] = {
+  final override def killExecutorsOnHost(host: String): Unit = {
     logInfo(s"Requesting to kill any and all executors on host ${host}")
-    killExecutors(scheduler.getExecutorsAliveOnHost(host).get.toSeq, replace = true, force = true)
+    driverEndpoint.send(KillExecutorsOnHost(host))
```

Comment:
I'd include a comment here on why you do delegate this to driver endpoint, rather than doing it immediately here, something along the lines of:

We have to be careful that there isn't a race between killing all executors on the bad host, and another executor getting registered on the same host.  We do that by doing it within the DriverEndpoint, which is guaranteed to handle one message at a time since its a ThreadSafeRPCEndpoint